src/service_aggregator.py

```diff
@@ -7,186 +7,168 @@
-        if coalesce_answer is None:
-            logger.error("Error detected: Got no answer from Answer coalesce, aborting.")
-            message['error'] = 'Error detected: Answer coalesce failed to return an answer, aborting.'
-            return message
-        # did we get a good response
-        elif len(coalesce_answer) == 0:
-            logger.error("Error detected: Got an empty answer from Answer coalesce, aborting.")
-            message['error'] = 'Error detected: Got an empty answer from Answer coalesce, aborting.'
-            return message
-        else:
+        coalesce_answer, status_code = post('coalesce', f'https://answercoalesce.renci.org/1.1/coalesce/{coalesce_type}', strider_answer)
+
+        # was any data returned
+        if len(coalesce_answer) == 0:
+            strider_answer['error'] = 'Answer coalesce error: Got empty response from strider.'
+            return strider_answer, status_code
+        # html error code returned, but at least there was something returned
+        elif status_code != 200:
+            logger.error(f'Answer coalesce error: HTML error status code {status_code} returned.')
+            coalesce_answer['error'] = f'Answer coalesce error: HTML error status code {status_code} returned.'
+            return coalesce_answer, status_code
+        # good html status code, but still do some checking
+        elif status_code == 200:
             logger.debug(f'coalesce out ({uid}): {json.dumps(coalesce_answer)}')
     else:
         # just use the strider result in Message format
         coalesce_answer: dict = strider_answer
 
     # call the omnicorp overlay service
-    omni_answer: dict = post('omnicorp', 'https://aragorn-ranker.renci.org/1.1/omnicorp_overlay', coalesce_answer)
+    omni_answer, status_code = post('omnicorp', 'https://aragorn-ranker.renci.org/1.1/omnicorp_overlay', coalesce_answer)
 
     # # open the tests file
     # with open(os.path.join(os.path.abspath(os.path.dirname(__file__)), '..', 'tests', 'omni_answer.json'), 'r') as tf:
     #     omni_answer = json.load(tf)
 
-    # was there an error getting data
-    if omni_answer is None:
-        logger.error('Error detected: Aragorn-ranker/omnicorp_overlay failed to return an answer, aborting.')
-        message['error'] = 'Error detected: Aragorn-ranker/omnicorp_overlay failed to return an answer, aborting.'
-        return message
-    # did we get a good response
-    elif len(omni_answer) == 0:
-        logger.error('Error detected: Got an empty answer from Aragorn-ranker/omnicorp_overlay, aborting.')
-        message['error'] = 'Error detected: Got an empty answer from Aragorn-ranker/omnicorp_overlay, aborting.'
-        return message
-    else:
-        logger.debug(f'omni out ({uid}): {json.dumps(omni_answer)}')
+    # was any data returned
+    if len(omni_answer) == 0:
+        coalesce_answer['error'] = 'Ranker/Omnicorp overlay error: Got empty response from strider.'
```

Comment:
Got empty response from omnicorp.  Not sure this is a valuable check, I can't think of a case where this would happen.

Reply:
i believe this could happen on an un-caught exception. i will remove this code as that wont happen again after the error enhancements are in place.

Reply:
resolved